src/Controls/tests/UITests/Tests/Issues/Issue17610.cs

```diff
@@ -31,17 +33,19 @@ public void RefreshIconDisappearsWhenUserCancelsRefreshByScrollingBackUp()
 			var rect1 = App.WaitForElement("Item4").GetRect();
 			var rect2 = App.WaitForElement("Item16").GetRect();
 
-			var fromX = (int)rect1.CenterX();
-			var fromY = (int)rect1.CenterY();
-			var toX = (int)rect2.CenterX();
-			var toY = (int)rect2.CenterY();
-
-			new TouchAction(androidApp.Driver)
-				.Press(fromX, fromY)
-				.MoveTo(toX, toY)
-				.MoveTo(fromX, fromY)
-				.Release()
-				.Perform();
+			int fromX = rect1.CenterX();
+			int fromY = rect1.CenterY();
+			int toX = rect2.CenterX();
+			int toY = rect2.CenterY();
+
+			OpenQA.Selenium.Appium.Interactions.PointerInputDevice touchDevice = new OpenQA.Selenium.Appium.Interactions.PointerInputDevice(PointerKind.Touch);
+			var refreshSequence = new ActionSequence(touchDevice, 0);
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, fromX, fromY, TimeSpan.Zero));
+			refreshSequence.AddAction(touchDevice.CreatePointerDown(PointerButton.TouchContact));
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, toX, toY, TimeSpan.FromMilliseconds(250)));
+			refreshSequence.AddAction(touchDevice.CreatePointerMove(CoordinateOrigin.Viewport, fromX, fromY, TimeSpan.FromMilliseconds(250)));
+			refreshSequence.AddAction(touchDevice.CreatePointerUp(PointerButton.TouchContact));
+			androidApp.Driver.PerformActions([refreshSequence]);
```

Comment:
Just out of curiosity, why is this better? Same question applies to all the other changes I guess 😄

Reply:
`TouchAction` is deprecated.